pkg/sensors/tracing/generickprobe.go

```diff
@@ -785,12 +786,18 @@ func addKprobe(funcName string, instance int, f *v1alpha1.KProbeSpec, in *addKpr
 		eventConfig.ArgM[a.Index] = uint32(argMValue)
 
 		argsBTFSet[a.Index] = true
-		argP := argPrinter{index: j, ty: argType, userType: userArgType, maxData: a.MaxData, label: a.Label}
+		argP := argPrinter{index: int(a.Index), ty: argType, userType: userArgType, maxData: a.MaxData, label: a.Label}
 		argSigPrinters = append(argSigPrinters, argP)
 
 		pathArgWarning(a.Index, argType, f.Selectors)
 	}
 
+	// Arguments are appended based on the index value,
+	// so the argument printers need to follow that
+	sort.Slice(argSigPrinters, func(i, j int) bool {
+		return argSigPrinters[i].index < argSigPrinters[j].index
+	})
```

Comment:
I've wrote some of my thoughts in: https://github.com/cilium/tetragon/issues/3711#issuecomment-2875584910.
One question I have is: if a user defines:
```
args:
 - index: 1
    type: ...
 - index: 0
   type: ...
```

What would be the order of the arguments in the ProcessKprobe event?